### ISSUE TITLE
Set non-unique non-primary column as foreign key

### ISSUE THREAD

**Issue by ysrinu** [susanBuck/dwa15-fall2017#184]
I have 'assets' table with 'id' column as PK, AI and mediumint(8), unsigned

![image](https://user-images.githubusercontent.com/25366909/33531183-9457d78c-d857-11e7-856f-ddf036a0cd2e.png)

I have a child table 'asset_repairs' with **no** PK or Unique key. There's a column asset_id that should correspond to the 'id' column of assets table. The reason there's no PK,AI key for this table is it will not be used for fetching anyway. All select queries will be based on asset_id.

I created plain index on column asset_id (non-unique). Now, mapping asset_id as foreign key on assets.id is failing both in phpmyadmin and laravel. If I can't make it work in phpmyadmin (native database) then it won't work from laravel anyway.

![image](https://user-images.githubusercontent.com/25366909/33531224-f90691aa-d857-11e7-9d72-a64ef801a788.png)
![image](https://user-images.githubusercontent.com/25366909/33531228-0b7ed00e-d858-11e7-973e-b09665a793a0.png)

When I fill the below form (asset_repairs) in phpMyAdmin

![image](https://user-images.githubusercontent.com/25366909/33531239-2cbb95d6-d858-11e7-82ef-8d721a8e390f.png)

with values:

![image](https://user-images.githubusercontent.com/25366909/33531246-4497376e-d858-11e7-81e0-efa4af746b3f.png)

and hit 'save' I get the below error:
`Error creating foreign key on asset_id (check data types)`

![image](https://user-images.githubusercontent.com/25366909/33531258-69815e24-d858-11e7-96da-833e07ca517e.png)

**Comment by ysrinu** [comment]
I figured out why.
The computers table also has column asset_id and the migration script created foreign key on this and the asset_repairs table was creating foreign key with the same name and hence didn't work. 
I resolved this by renaming the FK (suffixed with '**2**')
You would think phpMyAdmin would be intelligent enough to say a FK with that name already exists instead of that stupid message in the screenshot above. I lost precious time debugging this!

```
$table->index(["asset_id"], 'fk_asset_id_idx2);

$table->foreign('asset_id', 'fk_asset_id_idx2')
->references('id')->on('assets')
->onDelete('no action')
->onUpdate('no action');
```
